lib/MusicBrainz/Server/Form/Collection.pm

```diff
@@ -78,4 +78,19 @@ sub validate_type_id {
     }
 }
 
+sub validate_collaborators {
+    my $self = shift;
+
+    my @collaborators = $self->field('collaborators')->fields;
+    for my $collaborator (@collaborators) {
+        my $id_field = $collaborator->field('id');
+        my $name_field = $collaborator->field('name');
+        if (defined $name_field->value && !(defined $id_field->value)) {
+            return $id_field->add_error(
```

Comment:
It should probably add an error for all invalid collaborators, not just `return` the first one.

Reply:
True! Made the change.